test/integration/model/searchPath.test.js

```diff
@@ -0,0 +1,466 @@
+              where: {last_name: 'one'}, searchPath: SEARCH_PATH_ONE, include: [{
+                model: Restaurant, as: 'restaurant'
+              }]
+            });
+          }).then(function(obj) {
+            expect(obj).to.not.be.null;
+            expect(obj.restaurant).to.not.be.null;
+            expect(obj.restaurant.foo).to.equal('one');
+            return obj.getRestaurant({searchPath: SEARCH_PATH_ONE});
+          }).then(function(restaurant) {
+            expect(restaurant).to.not.be.null;
+            expect(restaurant.foo).to.equal('one');
+          });
+        });
+
+
+        it('should be able to insert and retrieve associated data into the table in schema_two', function() {
+          var Restaurant = this.Restaurant;
+          var Employee = this.Employee;
+          var restaurantId;
+
+          return Restaurant.create({
+            foo: 'two'
+          }, {searchPath: SEARCH_PATH_TWO}).then(function() {
+            return Restaurant.findOne({
+              where: {foo: 'two'}, searchPath: SEARCH_PATH_TWO
+            });
+          }).then(function(obj) {
+            expect(obj).to.not.be.null;
+            expect(obj.foo).to.equal('two');
+            restaurantId = obj.id;
+            return Employee.create({
+              first_name: 'Restaurant',
+              last_name: 'two',
+              restaurant_id: restaurantId
+            }, {searchPath: SEARCH_PATH_TWO});
+          }).then(function() {
+            return Restaurant.findOne({
+              where: {foo: 'two'}, searchPath: SEARCH_PATH_TWO, include: [{
+                model: Employee, as: 'employees'
+              }]
+            });
+          }).then(function(obj) {
+            expect(obj).to.not.be.null;
+            expect(obj.employees).to.not.be.null;
+            expect(obj.employees.length).to.equal(1);
+            expect(obj.employees[0].last_name).to.equal('two');
+            return obj.getEmployees({searchPath: SEARCH_PATH_TWO});
+          }).then(function(employees) {
+            expect(employees.length).to.equal(1);
+            expect(employees[0].last_name).to.equal('two');
+            return Employee.findOne({
+              where: {last_name: 'two'}, searchPath: SEARCH_PATH_TWO, include: [{
+                model: Restaurant, as: 'restaurant'
+              }]
+            });
+          }).then(function(obj) {
+            expect(obj).to.not.be.null;
+            expect(obj.restaurant).to.not.be.null;
+            expect(obj.restaurant.foo).to.equal('two');
+            return obj.getRestaurant({searchPath: SEARCH_PATH_TWO});
+          }).then(function(restaurant) {
+            expect(restaurant).to.not.be.null;
+            expect(restaurant.foo).to.equal('two');
+          });
+        });
+      });
+
+      describe('concurency tests', function() {
+        it('should build and persist instances to 2 schemas concurrently in any order', function() {
+          var Restaurant = this.Restaurant;
```

Comment:
@mickhansen This use case in particular fails when trying to use .schema() with the code below:

```
it('should but wont build and persist instances to 2 schemas concurrently in any order', function() {
          var Restaurant = this.Restaurant;

          var restaurauntModelSchema1 = Restaurant.schema('schema_one').build({bar: 'one.1'});
          var restaurauntModelSchema2 = Restaurant.schema('schema_two').build({bar: 'two.1'});

          return restaurauntModelSchema1.save()
            .then(function() {
              restaurauntModelSchema1 = Restaurant.schema('schema_one').build({bar: 'one.2'});
              return restaurauntModelSchema2.save();
            }).then(function() {
              return restaurauntModelSchema1.save();
            }).then(function() {
              return Restaurant.schema('schema_one').findAll();
            }).then(function(restaurantsOne) {
              expect(restaurantsOne).to.not.be.null;
              expect(restaurantsOne.length).to.equal(2);
              restaurantsOne.forEach(function(restaurant) {
                expect(restaurant.bar).to.contain('one');
              });
              return Restaurant.schema('schema_two').findAll();
            }).then(function(restaurantsTwo) {
              expect(restaurantsTwo).to.not.be.null;
              expect(restaurantsTwo.length).to.equal(1);
              restaurantsTwo.forEach(function(restaurant) {
                expect(restaurant.bar).to.contain('two');
              });
            });
        });
      });
```